arwen/elrondapi/bigFloatOps.go

```diff
@@ -0,0 +1,753 @@
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatCmp", v1_4_bigFloatCmp, C.v1_4_bigFloatCmp)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatAbs", v1_4_bigFloatAbs, C.v1_4_bigFloatAbs)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatSign", v1_4_bigFloatSign, C.v1_4_bigFloatSign)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatSqrt", v1_4_bigFloatSqrt, C.v1_4_bigFloatSqrt)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatPow", v1_4_bigFloatPow, C.v1_4_bigFloatPow)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatLog2", v1_4_bigFloatLog2, C.v1_4_bigFloatLog2)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatFloor", v1_4_bigFloatFloor, C.v1_4_bigFloatFloor)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatCeil", v1_4_bigFloatCeil, C.v1_4_bigFloatCeil)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatSetInt64", v1_4_bigFloatSetInt64, C.v1_4_bigFloatSetInt64)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatIsInt", v1_4_bigFloatIsInt, C.v1_4_bigFloatIsInt)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatSetBigInt", v1_4_bigFloatSetBigInt, C.v1_4_bigFloatSetBigInt)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatGetConstPi", v1_4_bigFloatGetConstPi, C.v1_4_bigFloatGetConstPi)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatGetConstE", v1_4_bigFloatGetConstE, C.v1_4_bigFloatGetConstE)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatSetBytes", v1_4_bigFloatSetBytes, C.v1_4_bigFloatSetBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	imports, err = imports.Append("bigFloatGetBytes", v1_4_bigFloatGetBytes, C.v1_4_bigFloatGetBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return imports, err
+}
+
+func oneIsInfinity(values ...*big.Float) bool {
+	for _, val := range values {
+		if val.IsInf() {
+			return true
+		}
+	}
+	return false
+}
+
+func allAreEqualToZero(values ...*big.Float) bool {
+	for _, val := range values {
+		if val.Sign() != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+//export v1_4_bigFloatNew
+func v1_4_bigFloatNew(context unsafe.Pointer, intBase, subIntBase, exponent int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatNew
+	metering.UseGas(gasToUse)
+
+	if exponent >= 0 {
+		_ = arwen.WithFault(arwen.ErrPositiveExponent, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return -1
+	}
+
+	value := float64(intBase) + float64(subIntBase)*math.Pow10(int(exponent))
```

Comment:
gas should depend on the length of exponent. Plus limit it to a certain constant

Reply:
by default if exponent < -323 , then Pow10(exponent) is equal to 0